src/CookieStorage.js

```diff
@@ -0,0 +1,70 @@
+
+import * as Cookies from "js-cookie";
+
+/** @class */
+export default class CookieStorage {
+
+  /**
+   * Constructs a new CookieStorage object
+   * @param {object} data Creation options.
+   * @param {string} data.domain Cookies domain (mandatory).
+   * @param {string} data.path Cookies path (default: '/')
+   * @param {integer} data.expires Cookie expiration (in days, default: 365)
+   */
+  constructor(data) {
+    this.domain = data.domain;
+    if (data.path) {
+      this.path = data.path;
+    } else {
+      this.path = '/';
+    }
+    if (data.expire){
+      this.expires = data.expires;
+    } else {
+      this.expires = 365;
+    }
+  }
+
+  /**
+   * This is used to set a specific item in storage
+   * @param {string} key - the key for the item
+   * @param {object} value - the value
+   * @returns {string} value that was set
+   */
+  setItem(key, value) {
+    Cookies.set(key, value, {path: this.path, expires: this.expires, domain: this.domain});
+    return Cookies.get(key);
+  }
+
+  /**
+   * This is used to get a specific key from storage
+   * @param {string} key - the key for the item
+   * This is used to clear the storage
+   * @returns {string} the data item
+   */
+  getItem(key) {
+    return Cookies.get(key)
+  }
+
+  /**
+   * This is used to remove an item from storage
+   * @param {string} key - the key being set
+   * @returns {string} value - value that was deleted
+   */
+  removeItem(key) {
+    return Cookies.remove(key, { path: this.path, domain: this.domain, secure: true});
```

Comment:
@marlontrapp , the `secure` property should be a constructor argument (for local development that doesn't have SSL).  A cookie can't be removed with `secure: true` if its been set with `secure: false`.

Reply:
I've added it as a constructor parameter now